core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -127,6 +131,19 @@ class ReplicaAlterLogDirsThread(name: String,
     logAppendInfo
   }
 
+  override def addPartitions(initialFetchStates: Map[TopicPartition, OffsetAndEpoch]): Set[TopicPartition] = {
+    partitionMapLock.lockInterruptibly()
+    try {
+      // It is possible that the log dir fetcher completed just before this call, so we
+      // filter only the partitions which still have a future log dir.
+      val filteredFetchStates = initialFetchStates.filterKeys(replicaMgr.futureLogExists)
```

Comment:
So there is a extra check of future folder here and the check is in the lock so the race condition is resolved.